dali/pipeline/data/tensor_vector.h

```diff
@@ -91,16 +91,12 @@ class DLL_PUBLIC TensorVector {
     return tensors_.cend();
   }
 
```

Comment:
I'd check the source code for patterns like:
`volume(tv.shape ())`, `tv.shape().num_elements()` and similar - perhaps we need the number of elements after all.

Reply:
Perhaps, but in this PR I aim at removing APIs before merging them, we can start cluttering them again after that (I wish we didn't).